### PR TITLE
add distclean-tmp target to core

### LINKED ISSUE #356: Remove $(ERLANG_MK_TMP) directory on "make distclean"
Looks like it's not done yet. Oops!

### PR DESCRIPTION
**Pull request by ingwinlu**
Removes $(ERLANG_MK_TMP) directory on make distclean.

resolves #356

### CODE REVIEW

**Comment by essen** [comment]
Any chance you can also add a test? Should be more or less what core-clean-crash-dump does in test/Makefile, except with distclean and a different file check (you can create the folder manually if building doesn't do it, it probably won't for an empty app). :-)

File: test/Makefile [hunk 1]
```diff
@@ -88,12 +88,13 @@ $(eval $(foreach t,$(patsubst %.mk,%,$(patsubst plugin_%,%,$(wildcard plugin_*.m
 
 # Tests that don't easily fit into other categories.
 
-core:: core-clean-crash-dump core-help
+core:: core-clean-crash-dump core-clean-tmp core-help
```
Comment (essen) [inline test/Makefile]: Better call it core-distclean-tmp.

File: test/Makefile [hunk 2]
```diff
@@ -88,12 +88,13 @@ $(eval $(foreach t,$(patsubst %.mk,%,$(patsubst plugin_%,%,$(wildcard plugin_*.m
...
 clean-core:: clean-core-clean-crash-dump clean-core-help
 
-.PHONY: core-clean-crash-dump core-help clean-core-clean-crash-dump clean-core-help
+.PHONY: core-clean-crash-dump core-clean-tmp core-help \
+    clean-core-clean-crash-dump clean-core-clean-tmp clean-core-help
```
Comment (essen) [inline test/Makefile]: No need to split the line.

File: test/Makefile [hunk 3]
```diff
@@ -112,6 +113,22 @@ core-clean-crash-dump: build clean-core-clean-crash-dump
...
+
+	$i "Bootstrap a new OTP application named $(APP)"
+	$t mkdir $(APP)/
+	$t cp ../erlang.mk $(APP)/
+	$t $(MAKE) -C $(APP) -f erlang.mk bootstrap all $v
+
+	$i "Check that a .erlang.mk directory exists"
+	$t test -d $(APP)/.erlang.mk
+
+	$i "Clean the application"
```
Comment (essen) [inline test/Makefile]: Distclean the application

File: test/Makefile [hunk 4]
```diff
@@ -112,6 +113,22 @@ core-clean-crash-dump: build clean-core-clean-crash-dump
...
+	$t $(MAKE) -C $(APP) distclean $v
+
+	$i "Check if .erlang.mk directory got removed"
+	$t test ! -d $(APP)/.erlang.mk
```
Comment (essen) [inline test/Makefile]: test ! -e

Otherwise if there's a normal file named .erlang.mk it'll succeed. Nitpick. :-)

File: core/core.mk [hunk 5]
```diff
@@ -12,7 +12,8 @@
 # ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 # OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-.PHONY: all app deps search rel docs install-docs check tests clean distclean help erlang-mk
+.PHONY: all app deps search rel docs install-docs check tests clean \
```
Comment (ingwinlu) [inline core/core.mk]: should I remove the linesplit here as well?

Comment (essen) [inline core/core.mk]: Yep, thanks.

File: test/Makefile [hunk 6]
```diff
@@ -88,12 +88,13 @@ $(eval $(foreach t,$(patsubst %.mk,%,$(patsubst plugin_%,%,$(wildcard plugin_*.m
 
 # Tests that don't easily fit into other categories.
 
-core:: core-clean-crash-dump core-help
+core:: core-clean-crash-dump core-clean-tmp core-help
 clean-core:: clean-core-clean-crash-dump clean-core-help
```
Comment (ingwinlu) [inline test/Makefile]: missed cleanup of new target here as well, whoops

**Pull request merged**